src/swarm/neo/request/RequestEventDispatcher.d

```diff
@@ -629,39 +634,56 @@ public struct RequestEventDispatcher
 
     private void register ( MessageFiber fiber, EventRegistration event )
     {
-        with ( event.Active ) switch ( event.active )
+        WaitingFiber* in_list;
+
+        // See if a matching element is already in this.waiting_fibers.
+        foreach ( ref waiting_fiber; this.waiting_fibers )
         {
-            case message:
-                foreach ( waiting_fiber; this.waiting_fibers )
-                    if ( waiting_fiber.event.active == message )
+            if ( waiting_fiber.fiber == fiber && waiting_fiber.event == event )
+            {
+                assert(in_list is null);
+                assert(!waiting_fiber.enabled);
+                in_list = &waiting_fiber;
+                break;
+            }
+
+            // Ensure the user isn't trying to do anythign crazy.
+            if ( waiting_fiber.enabled &&
```

Comment:
Hmm I notice that the main loop to find the matching element can `break`, which means that the sanity check loop doesn't complete. These should probably be two separate loops now.

Reply:
Or don't `break` and set `in_list` only if it is not `null`.

Reply:
Updated.

Reply:
Updated with David's suggestion.

Reply:
Is `assert(in_list is null)` still correct now that this loop doesn't `break`?

Reply:
Yes, because it should be impossible to register the same fiber/event pair twice.

Reply:
(That assertion actually came from an earlier version of the code which didn't break the loop.)